arangodb-net-standard/CollectionApi/CollectionApiClient.cs

```diff
@@ -128,21 +128,45 @@ public async Task<GetCollectionResponse> GetCollectionAsync(string collectionNam
                 throw await GetApiErrorException(response);
             }
         }
-        
+
         /// Read properties of a collection
         /// /_api/collection/{collection-name}/properties
         /// </summary>
         /// <param name="options"></param>
         /// <returns></returns>
         public async Task<GetCollectionPropertiesResponse> GetCollectionPropertiesAsync(string collectionName)
         {
-            var response = await _transport.GetAsync(_collectionApiPath + "/" + collectionName + "/properties");            
-            if (response.IsSuccessStatusCode)
+            using (var response = await _transport.GetAsync(_collectionApiPath + "/" + collectionName + "/properties"))
             {
-                var stream = await response.Content.ReadAsStreamAsync();
-                return DeserializeJsonFromStream<GetCollectionPropertiesResponse>(stream);
+                if (response.IsSuccessStatusCode)
+                {
+                    var stream = await response.Content.ReadAsStreamAsync();
+                    return DeserializeJsonFromStream<GetCollectionPropertiesResponse>(stream);
+                }
+                throw await GetApiErrorException(response);
+            }
+        }
+
+        /// <summary>
+        /// Rename a collection.
+        /// /_api/collection/{collection-name}/rename
+        /// </summary>
+        /// <param name="collectionName"></param>
+        /// <param name="request"></param>
+        /// <returns></returns>
+        public async Task<RenameCollectionResponse> RenameCollectionAsync(string collectionName, RenameCollectionRequest request)
+        {
+            StringContent content = GetStringContent(request, true, false);
+            using (var response = await _transport.PutAsync(_collectionApiPath + "/" + collectionName + "/rename", content))
+            {
+                if (response.IsSuccessStatusCode)
+                {
+                    var stream = await response.Content.ReadAsStreamAsync();
+                    var collection = DeserializeJsonFromStream<RenameCollectionResponse>(stream, false, true);
```

Comment:
This seems to be different from what we have been doing up to now. Any reason you used `(stream, false, true)` instead of `(stream, true)`? I'm not sure what's the impact of ignoring `null` values, as well as not using `camelCase`, when deserializing. @rossmills99 maybe you know about that?

Reply:
Tested with either options and you get the same result and the test passes. So I am not sure which configuration to use?

Reply:
Hmm... well this is embarassing.  I did an experiment by commenting out the code that uses these options:
```
protected T DeserializeJsonFromStream<T>(Stream stream, bool useCamelCasePropertyNames = false, bool ignoreNullValues = false)
        {
            if (stream == null || stream.CanRead == false)
            {
                return default(T);
            }

            using (var sr = new StreamReader(stream))
            using (var jtr = new JsonTextReader(sr))
            {
                var js = new JsonSerializer();
                //{
                //    NullValueHandling = ignoreNullValues ? NullValueHandling.Ignore : NullValueHandling.Include
                //};
                //if (useCamelCasePropertyNames)
                //{
                //    js.ContractResolver = new CamelCasePropertyNamesContractResolver();
                //}
                var searchResult = js.Deserialize<T>(jtr);
                return searchResult;
            }
        }
```

All the tests still pass.

I have to conclude that the options only apply when serializing, not deserializing, which on reflection makes sense.  I think when deserializing, camel-case property names are automatically mapped to PascalCase property names.  And the "ignore null values" option makes no sense whatsoever in a deserialization context.

So I guess the use of these options has been unnecessary from the start?  I think we might be able to just remove them from the code.

Reply:
I think you're right. We can probably create a new issue to remove this.